### PR TITLE
KD-3404: Self check in without log in

### PR DESCRIPTION
**Pull request by juhak18**
Borrower can return items without log in

### CODE REVIEW

**Comment by kohaputti** [comment]
Can you submit this to community first?

**Comment by johannaraisa** [comment]
I agree with @kohaputti, we should try to get this into community version. Get their opinion about it and atleast get a bug number from bugzilla.

**Comment by juhak18** [comment]
Maybe I should first put it in the production version of Oulu. It's not tested in production yet. We need this feature soon.

**Comment by johannaraisa** [comment]
I have a plan to update productions again with latest master. I will add it for testing today, could this wait until next week?

**Comment by juhak18** [comment]
It can wait one or two weeks. Not that urgent.

**Comment by kohaputti** [comment]
@juhak18 how are you planning to progress with this? Will you work on the community version first or do you plan to submit a v2 for this only here? I'm asking this so I know whether to close this or keep open for the v2.

For v2 remove the unnecessary formatting from the patch so rebases and reverts will be a lot easier.

**Comment by juhak18** [comment]
We are going to put this to production soon. If it works here in Oulu, then can I offer it to community.

File: koha-tmpl/opac-tmpl/bootstrap/en/modules/sco/sco-main.tt [hunk 1]
```diff
@@ -307,23 +310,64 @@
...
                                         <input type="hidden" name="op" value="login" />
                                     </fieldset>
                                 </form>
+                               
+                               </td><td style="padding:20px">
+
+                               <form id="barcodeform" action="/cgi-bin/koha/sco/sco-main.pl" name="barcodeform" method="post">
+                                    <fieldset class="checkout brief">
+                                        [% IF ( authbylogin ) %]
+                                            <input type="hidden" name="reserve_id" value="[% reserve_id %]" />
```
Comment (kohaputti) [inline koha-tmpl/opac-tmpl/bootstrap/en/modules/sco/sco-main.tt]: This doesn't seem to be used at all. Remove.

File: opac/sco/sco-main.pl [hunk 2]
```diff
@@ -94,49 +101,142 @@
 }
 $template->param(AllowSelfCheckReturns => $allowselfcheckreturns);
 
-
 my $issuerid = $loggedinuser;
-my ($op, $patronid, $patronlogin, $patronpw, $barcode, $confirmed) = (
+my ($op, $patronid, $patronlogin, $patronpw, $barcode, $confirmed, $uibarcode,$checkinmessage,
+    $reserve_id) = (
```
Comment (kohaputti) [inline opac/sco/sco-main.pl]: Remove $reserved_id here, too

File: opac/sco/sco-main.pl [hunk 3]
```diff
@@ -94,49 +101,142 @@
...
     $query->param("op")         || '',
     $query->param("patronid")   || '',
     $query->param("patronlogin")|| '',
     $query->param("patronpw")   || '',
     $query->param("barcode")    || '',
     $query->param("confirmed")  || '',
+    $query->param("uibarcode")  || '',
+    $query->param("checkinmessage") || '',
+    $query->param("reserve_id") || '',
```
Comment (kohaputti) [inline opac/sco/sco-main.pl]: Remove reserved_id here, too

File: opac/sco/sco-main.pl [hunk 4]
```diff
@@ -94,49 +101,142 @@
...
+        my $diffBranchSend = ($branch ne $diffBranchReturned) ? $diffBranchReturned : undef;
+        my $iteminfo   = GetBiblioFromItemNumber($itemnumber);
+
+        if($diffBranchSend) {
+            ModReserveAffect( $itemnumber, $resborrower, $diffBranchSend, $reserve_id);
+            ModItemTransfer($itemnumber,$branch,$diffBranchReturned);         
+        }
+        else {
+            $settransit = C4::Context->preference('RequireSCCheckInBeforeNotifyingPickups');
+            $settransit = 0 unless $settransit;
```
Comment (kohaputti) [inline opac/sco/sco-main.pl]: Just do this in the variable definition above with
 ```my $settransit = C4::Context->preference('RequireSCCheckInBeforeNotifyingPickups') ? 1 : 0;```.

File: opac/sco/sco-main.pl [hunk 5]
```diff
@@ -94,49 +101,142 @@
...
+       if ($biblio) {
+           $checkinmessage = "Returned ".$biblio->{title};
+       }
+       if ($checkinmember) {
+          $checkinbranchcode = $checkinmember->{branchcode};
+
+          #home branch of item
+          if ($checkinbranchcode) {
+             my $tmpdbh = C4::Context->dbh;
+             my $tmpsth = $tmpdbh->prepare("select branchname from branches where branchcode = ? ");
```
Comment (kohaputti) [inline opac/sco/sco-main.pl]: Use already existing code for this: `Koha::Libraries->find()`

**Pull request closed without merging**